File: frontend/ui/src/app/types/secret.ts
import {BaseModel, UserAccount} from '@distr-sh/distr-sdk';

export interface Secret extends BaseModel {
  id: string;
  createdAt: string;
  updatedAt: string;
  updatedByUserAccountId?: string;
  customerOrganizationId?: string;
  key: string;
  updatedBy?: UserAccount;
}

File: internal/types/secret.go
package types

import (
	"time"

	"github.com/distr-sh/distr/internal/dbcrypto"
	"github.com/google/uuid"
)

type Secret struct {
	ID                     uuid.UUID       `db:"id"`
	CreatedAt              time.Time       `db:"created_at"`
	UpdatedAt              time.Time       `db:"updated_at"`
	UpdatedByUserAccountID *uuid.UUID      `db:"updated_by_useraccount_id"`
	OrganizationID         uuid.UUID       `db:"organization_id"`
	CustomerOrganizationID *uuid.UUID      `db:"customer_organization_id"`
	Key                    string          `db:"key"`
	Value                  dbcrypto.String `db:"value"`
}

type SecretWithUpdatedBy struct {
	Secret
	UpdatedBy *UserAccount `db:"updated_by"`
}
